control-plane/agents/src/bin/ha/node/server.rs

```diff
@@ -65,24 +65,37 @@ impl NodeAgentSvc {
 }
 
 /// Disconnect cached NVMe controller.
-fn disconnect_controller(ctrlr: &NvmeController) -> Result<(), SvcError> {
+fn disconnect_controller(ctrlr: &NvmeController, new_path: String) -> Result<(), SvcError> {
+    let parsed_path = parse_uri(new_path.as_str()).ok_or(SvcError::InvalidArguments {})?;
+
+    let host = parsed_path.0;
+    let port = parsed_path.1;
+
     match get_nvme_path_buf(&ctrlr.path) {
         Some(pbuf) => {
             let subsystem = Subsystem::new(pbuf.as_path()).map_err(|_| SvcError::Internal {
                 details: "Failed to get NVMe subsystem for controller".to_string(),
             })?;
 
-            tracing::info!(
+            if subsystem.address == format!("traddr={},trsvcid={}", host, port) {
```

Comment:
Can we add a new type SubsystemAddr, and here use SubsystemAddr::new(host, port) ?

Reply:
Yeah, that would be good. Thanks